js/ad_manager_vast.js

```diff
@@ -851,14 +1047,19 @@ OO.Ads.manager(function(_, $) {
      * @returns {boolean} True if the ad was added to the timeline successfully, false otherwise.
      */
     var addToTimeline = _.bind(function(metadata) {
+      if (!metadata) return;
       var timeline = [];
       var ad = generateAd(metadata);
+      var isVPaid = metadata.data && metadata.data.adType === 'vpaid';
 
-      if (metadata.streamUrl != null || (ad.adType == this.amc.ADTYPE.LINEAR_VIDEO && !_.isEmpty(metadata.streams))) {
+      if (metadata.streamUrl != null ||
+          (ad.adType == this.amc.ADTYPE.LINEAR_VIDEO && !_.isEmpty(metadata.streams)) ||
+          (ad.adType === this.amc.ADTYPE.NONLINEAR_OVERLAY && !_.isEmpty(metadata.data.nonLinear.mediaFiles.url))) {
```

Comment:
This might need to get integrated with Doug's error handling changes. I recall errors for when streams or media files aren't defined. We need to check with Doug on this when we merge.